include/deal.II/grid/grid_tools.h

```diff
@@ -3704,10 +3713,14 @@ namespace GridTools
 
 
 
-  template <int dim, typename Predicate, int spacedim>
-  void
-  transform(const Predicate &             predicate,
-            Triangulation<dim, spacedim> &triangulation)
+  template <int dim, typename Transformation, int spacedim>
+  DEAL_II_CXX20_REQUIRES(
+    (std::invocable<Transformation, Point<spacedim>> &&
+     std::assignable_from<
+       Point<spacedim> &,
+       std::invoke_result_t<Transformation, Point<spacedim>>>))
+  void transform(const Transformation &        transformation,
+                 Triangulation<dim, spacedim> &triangulation)
```

Comment:
Same issue as https://github.com/dealii/dealii/pull/14933/files#r1143909567 with regard to the name "Predicate".